lib/liquid/combinators/tags/cycle.ex

```diff
@@ -0,0 +1,73 @@
+defmodule Liquid.Combinators.Tags.Cycle do
+  @moduledoc """
+  Implementation of `cycle` tag. Can be named or anonymous, rotates through pre-set values
+  Cycle is usually used within a loop to alternate between values, like colors or DOM classes.
+  ```
+    {% for item in items %}
+    <div class="{% cycle 'red', 'green', 'blue' %}"> {{ item }} </div>
+    {% end %}
+  ```
+  ```
+    <div class="red"> Item one </div>
+    <div class="green"> Item two </div>
+    <div class="blue"> Item three </div>
+    <div class="red"> Item four </div>
+    <div class="green"> Item five</div>
+  ```
+  Loops through a group of strings and outputs them in the order that they were passed as parameters.
+  Each time cycle is called, the next string that was passed as a parameter is output.
+  cycle must be used within a for loop block.
+  Input:
+  ```
+    {% cycle 'one', 'two', 'three' %}
+    {% cycle 'one', 'two', 'three' %}
+    {% cycle 'one', 'two', 'three' %}
+    {% cycle 'one', 'two', 'three' %}
+  ```
+  Output:
+  ```
+    one
+    two
+    three
+    one
+  ```
+  """
+  import NimbleParsec
+  alias Liquid.Combinators.{Tag, General}
+
+  @type t :: [cycle: [group: String.t(), values: [LexicalToken.value()]]]
```

Comment:
Shouldn't a Cycle type be just a tagged tuple instead of a list of tagged tuples?

Reply:
Every tag is a keyword list (which is a list of tagged tuples)